### ISSUE TITLE
load_all_vars throws IndexError

### ISSUE THREAD

**Issue by julibeg** [mgalardini/pyseer#151]
It can happen that the last samples don't have a single 'active' variant after variant filtering. In such a case, since no `shape` parameter is supplied to `csr_matrix` here 
https://github.com/mgalardini/pyseer/blob/23481c2dbe0999ebef4ee5a61d54c8358254a61e/pyseer/enet.py#L113
the sparse matrix has too small dimensions (`variants.shape[1]` is smaller than `len(sample_order)`), which triggers an IndexError downstream. To fix this, simply add `shape=(len(selected_vars), len(sample_order))` to the call to `csr_matrix` so that 
```
variants = csr_matrix((data, indices, indptr), dtype=float, shape=(len(selected_vars), len(sample_order)))
```
Should I open a PR?

**Comment by johnlees** [comment]
Thanks for spotting this, yes this is an oversight and we should definitely be giving the shape argument! If you want to open a PR that'd be really helpful, and I can merge right away

**Comment by julibeg** [comment]
done

**Comment by mgalardini** [comment]
Thanks a lot! @johnlees do you think this change warrants a new version once merged?

**Comment by johnlees** [comment]
Merged, but one of the unit tests now failing:
```
    def test_load_all_vars_rtab(self):
        p = pd.read_csv(P,
                        index_col=0,
                        sep='\t')['binary']
        infile, sample_order = open_rtab(PRES)
        variants, sidx, vidx  = load_all_vars('Rtab', p, False, None,
                                              infile, set(p.index),
                                              sample_order,
                                              0.25, 0.75, 1.0, False)
        self.assertEqual(variants.shape, (7, 50))
        self.assertEqual(variants.sum(), 103.0)
        self.assertTrue(abs(variants.toarray()[0] -
                            np.array([0., 0., 1., 0., 0., 0., 1., 0., 1., 0.,
                                      0., 0., 1., 0., 0., 0., 0.,
                                      0., 1., 0., 0., 0., 0., 0., 0., 1., 0.,
                                      0., 1., 1., 0., 0., 0., 0.,
                                      0., 0., 0., 0., 1., 1., 0., 1., 0., 1.,
                                      0., 1., 0., 0., 1., 0.
                                      ])).max() < 1E-7)
        self.assertEqual(len(sidx), 7)
        self.assertEqual(sidx, [1426, 1436, 1463, 1484, 1492, 1496, 1498])
        self.assertEqual(vidx, 1499)
        # too few OGs
        with self.assertRaises(ValueError):
            infile, sample_order = open_rtab(PRESSMALL, compressed=False)
>           _  = load_all_vars('Rtab', p, False, None,
                               infile, set(p.index), sample_order,
                               0.01, 0.99, 1.0, False)
E           AssertionError: ValueError not raised
```

**Comment by johnlees** [comment]
Do we need an:
```
if len(selected_vars) == 0:
    raise ValueError
```
at the end of `load_all_vars`

**Comment by johnlees** [comment]
Yep, fixed in 0c34ce091aec9ec488cbae433c726ab936f50ae5
